dataprofiler/profilers/base_column_profilers.py

```diff
@@ -36,12 +36,12 @@ def __init__(self, name: Optional[str]) -> None:
         :param name: Name of the dataset
         :type name: String
         """
-        self.name = name
+        self.name: Optional[str] = name
```

Comment:
I'm not sure this is allowed to be optional. Reason for choosing `Optional` here?

Reply:
nvm it is the tests, right?

Reply:
Some of the subclasses initialize the profiler with `None`. For example, `categorical_column_profile.py:66` is `merged_profile = CategoricalColumn(None)`.

Reply:
Yup, makes sense.